src/transformers/models/llama/modeling_llama.py

```diff
@@ -123,7 +123,9 @@ def forward(self, x, position_ids, seq_len=None):
         # x: [bs, num_attention_heads, seq_len, head_size]
         inv_freq_expanded = self.inv_freq[None, :, None].float().expand(position_ids.shape[0], -1, 1)
         position_ids_expanded = position_ids[:, None, :].float()
-        freqs = (inv_freq_expanded @ position_ids_expanded).transpose(1, 2)
+        # Force float32 since bfloat16 loses precision on long contexts
+        with torch.autocast(device_type=position_ids_expanded.device.type, enabled=False):
```

Comment:
Do you think this adds slow downs otherwise?

Reply:
I think it will in general add some overhead, since before this was done in float16, but now its float32. For inference, the hit will be probably a bit bigger, whilst training the rest of the ops trump over the matrix multiply.

The overhead though should be minute, I would assume 5% to 15% extra time from a back of the envelope calculation. Another issue is VRAM use will be slightly higher

Reply:
Overhead because of float32 is alright, I meant the context manager rather than forcing float32! 
🤗

Reply:
OOO :))) For inference - tbh unsure - I'm assuming `torch.compile` will optimize it out. For training - should be negligible :)